src/test/scala/com/databricks/labs/mosaic/test/MosaicSpatialQueryTest.scala

```diff
@@ -0,0 +1,136 @@
+package com.databricks.labs.mosaic.test
+
+import com.databricks.labs.mosaic.core.geometry.api.GeometryAPI
+import com.databricks.labs.mosaic.core.geometry.api.GeometryAPI.{ESRI, JTS}
+import com.databricks.labs.mosaic.functions.MosaicContext
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import org.apache.spark.sql.catalyst.plans.PlanTest
+import com.databricks.labs.mosaic.core.index.{BNGIndexSystem, H3IndexSystem, IndexSystem}
+import org.apache.spark.sql.catalyst.expressions.CodegenObjectFactoryMode
+import org.apache.spark.sql.functions.col
+import org.apache.spark.sql.internal.SQLConf
+import org.scalatest.{Assertions, BeforeAndAfterEach, Suite, Tag}
+
+abstract class MosaicSpatialQueryTest extends PlanTest with MosaicHelper {
```

Comment:
same here, can we get some scaladocs in place for usability.
And same for all 4 API methods.

Reply:
It sounds a bit weird to me we call this MosaicSpatialQueryTest and we are extending PlanTest.
Any specific reason for switching from QueryTest to PlanTest.

Reply:
I added some docs! The naming was inspired by spark (I used essentially the same approach as spark (https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/QueryTest.scala) and I'm struggling to find a better alternative. Let me know if you think of something, I'm open to ideas!